ACPs/99-validatorsetmanager-contract/README.md

```diff
@@ -0,0 +1,264 @@
+```text
+ACP: 99
+Title: ValidatorSetManager Solidity Contract
+Author(s): Gauthier Leonard <https://github.com/Nuttymoon>
+Discussions-To: https://github.com/avalanche-foundation/ACPs/discussions/98
+Status: Proposed
+Track: Best Practices
+Dependencies: 77
+```
+
+## Abstract
+
+Define (i) a reference implementation for a minimal Solidity smart contract to manage a Subnet’s validator set from an Avalanche EVM chain, as well as (ii) a reference architecture to easily plug in custom “security modules” on top of this contract (e.g. to implement a PoS Subnet).
+
+The proposed name for this contract is `ValidatorSetManager`.
+
+This proposal should also help define the payload of the P-Chain notifications sent to the Subnet upon validator addition, weight change and removal.
+
+This ACP relies on concepts introduced in [ACP-77 (Reinventing Subnets)](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets). It depends on it to be marked as `Implementable`.
+
+## Motivation
+
+[ACP-77 (Reinventing Subnets)](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets) opens the door to managing a Subnet validator set (stored on the P-Chain) from any chain on the Avalanche Network. The P-Chain allows a Subnet to specify a Warp Derived Address as the validator set manager. The `(blockchainID, address)` pair specified by the Warp Derived Address is responsible for sending Warp `AddressedCall` payloads to issue `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` on the P-Chain. This enables an onchain program add, modify the weight of, and remove validators.
```

Comment:
In ACP77, don't think there will be a `SetSubnetValidatorWeightTx`, instead this would be the specified tuple (chainID, address) in `ConvertSubnetTx`

Reply:
https://github.com/avalanche-foundation/ACPs/blob/11e8880a839c4b2020c7c84423c6c9042092487f/ACPs/77-reinventing-subnets/README.md?plain=1#L205

Reply:
my bad commented on the wrong line, meant `SetSubnetValidatorManagerTx` [here](https://github.com/avalanche-foundation/ACPs/pull/99/files#diff-a7a0317424a0341b7a5426c1c751b11b8683adf4e990c0d1f37c45641459d898R36) is not defined in ACP77, and will instead be set in `ConvertSubnetTx`.